main_test.go

```
@@ -0,0 +1,1482 @@
+		t.Fatal("test timed out waiting for shutdown to be triggered")
+	}
+}
+
+// ---------------------------------------------------------------------------
+// E2E tests — httptest.NewServer wrapping handleRequest
+// ---------------------------------------------------------------------------
+
+// doRequest sends a GET to the proxy with the configured test hostname.
+func doRequest(t *testing.T, proxyURL string) *http.Response {
+	t.Helper()
+	req, err := http.NewRequest("GET", proxyURL, nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)